hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaMessagesProducer.java

```diff
@@ -0,0 +1,60 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResult;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducer;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingObserver;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults;
+import pl.allegro.tech.hermes.frontend.publishing.PublishingCallback;
+import pl.allegro.tech.hermes.frontend.publishing.message.Message;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.List;
+
+@Singleton
+public class KafkaMessagesProducer implements BrokerMessagesProducer {
```

Comment:
I think it's quite hard to differentiate this class from `KafkaBrokerMessageProducer` without peeking to both classes. It would be better to name this class accordingly to the role its playing, just like `InternalKafkaMessageBatchProducer`. It is only example name. It should saying: "This class is for our internal kafka healthchecking monitornig"